### PR TITLE
Test comments cause violations when they should

### LINKED ISSUE #208: test that violation still happens when commented if commentDoesntViolate is false in verifyRule
Thanks @segiddins for the idea in https://github.com/realm/SwiftLint/pull/203#discussion_r45130602

### PR DESCRIPTION
**Pull request by segiddins**
Closes #208

### CODE REVIEW

File: Source/SwiftLintFrameworkTests/StringRuleTests.swift [hunk 1]
```diff
@@ -46,7 +46,7 @@ class StringRuleTests: XCTestCase {
     }
 
     func testFileShouldntStartWithWhitespace() {
-        verifyRule(LeadingWhitespaceRule.description, commentDoesntViolate: false)
+        verifyRule(LeadingWhitespaceRule.description, commentDoesntViolate: true)
```
Comment (jpsim) [inline Source/SwiftLintFrameworkTests/StringRuleTests.swift]: the parameter has a default value of `true` so you can just omit it.

Comment (segiddins) [inline Source/SwiftLintFrameworkTests/StringRuleTests.swift]: can we add a rule for that?

File: Source/SwiftLintFrameworkTests/TestHelpers.swift [hunk 2]
```diff
@@ -33,9 +33,18 @@ extension XCTestCase {
 
         if commentDoesntViolate {
             let commentedViolations = ruleDescription.triggeringExamples.flatMap {
-                violations("/** " + $0, ruleDescription)
+                violations("/**\n  " + $0 + "\n */", ruleDescription)
             }
             XCTAssertEqual(commentedViolations, [])
+        } else {
+            let commentedViolations = ruleDescription.triggeringExamples.flatMap {
```
Comment (jpsim) [inline Source/SwiftLintFrameworkTests/TestHelpers.swift]: you could reuse the `commentedViolations` and `XCTAssertEqual(commentedViolations, Array(count: commentDoesntViolate  ? 0 : ruleDescription.triggeringExamples.count, repeatedValue: ruleDescription)`.

**Comment by jpsim** [comment]
:+1: thanks Sam!

**Pull request merged**